tools/stress-cluster/scripts/deploy_stress_tests.ps1

```diff
@@ -0,0 +1,160 @@
+function DeployStressTests(
+    [string]$searchDirectory = '.',
+    [hashtable]$filters = @{},
+    [string]$environment = 'test',
+    [string]$repository = 'images',
+    [boolean]$pushImages = $false,
+    [string]$clusterGroup = 'rg-stress-test-cluster-',
+    [string]$deployId = 'local',
+    [string]$subscription = 'Azure SDK Test Resources'
+) {
+    if ($PSCmdlet.ParameterSetName -eq 'DoLogin') {
+        Login $subscription $clusterGroup $pushImages
+    }
+
+    RunOrExit helm repo add stress-test-charts https://stresstestcharts.blob.core.windows.net/helm/
+    Run helm repo update
+    if ($LASTEXITCODE) { return $LASTEXITCODE }
+
+    $pkgs = FindStressPackages $searchDirectory $filters
+    Write-Output "" "Found $($pkgs.Length) stress test packages:"
+    Write-Output $pkgs.Directory ""
+    foreach ($pkg in $pkgs) {
+        Write-Output "Deploying stress test at '$($pkg.Directory)'"
+        DeployStressPackage $pkg $deployId $environment $repository $pushImages
+    }
+
+    Write-Output "Releases deployed by $deployId"
+    Run helm list --all-namespaces -l deployId=$deployId
+
+    if ($FailedCommands) {
+        Write-Warning "" "The following commands failed:" ""
+        foreach ($cmd in $FailedCommands) {
+            Write-Error "'$($cmd.command)' failed with code $($cmd.code)" -ErrorAction 'Continue'
+        }
+        exit 1
+    }
+}
+
+function DeployStressPackage(
+    [object]$pkg,
+    [string]$deployId,
+    [string]$environment,
+    [string]$repository,
+    [boolean]$pushImages
+) {
+    $registry = (az acr list -g $clusterGroup -o json | ConvertFrom-Json).name
+    if (!$registry) {
+        Write-Output "Could not find container registry in resource group $clusterGroup"
+        exit 1
+    }
+
+    if ($pushImages) {
+        Run helm dependency update $pkg.Directory
+        if ($LASTEXITCODE) { return $LASTEXITCODE }
+
+        $dockerFiles = Get-ChildItem "$($pkg.Directory)/Dockerfile*"
+        foreach ($dockerFile in $dockerFiles) {
+            # Infer docker image name from parent directory name, if file is named `Dockerfile`
+            # or from suffix, is file is named like `Dockerfile.myimage` (for multiple dockerfiles).
+            $prefix, $imageName = $dockerFile.Name.Split(".")
+            if (!$imageName) {
+                $imageName = $dockerFile.Directory.Name
+            }
+            $imageTag = "$registry.azurecr.io/$($repository.ToLower())/$($imageName):$deployId"
+            Write-Output "Building and pushing stress test docker image '$imageTag'"
+            Run docker build -t $imageTag -f $dockerFile.FullName $dockerFile.DirectoryName
+            if ($LASTEXITCODE) { return $LASTEXITCODE }
+            Run docker push $imageTag
+            if ($LASTEXITCODE) { return $LASTEXITCODE }
+        }
+    }
+
+    Write-Output "Creating namespace $($pkg.Namespace) if it does not exist..."
+    kubectl create namespace $pkg.Namespace --dry-run=client -o yaml | kubectl apply -f -
```

Comment:
Should you use one of your run helpers here?

Reply:
I was having trouble dealing with `>` and `|` operators when passing them through to `Run()`, so there are a few places in the script where I bypass it to keep `Run()` simple. In all places the subsequent command will fail if the previous command fails and is not caught, however.